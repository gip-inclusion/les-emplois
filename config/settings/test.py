import os


ITOU_ENVIRONMENT = "DEV"
os.environ["ITOU_ENVIRONMENT"] = ITOU_ENVIRONMENT

from .base import *  # noqa: E402,F403


SECRET_KEY = "foobar"
ALLOWED_HOSTS = []

# We *want* to do the same `collectstatic` on the CI than on PROD to catch errors early,
# but we don't want to do it when running the test suite locally for performance reasons.
if not os.getenv("CI", False):
    # `ManifestStaticFilesStorage` (used in base settings) requires `collectstatic` to be run.
    STORAGES["staticfiles"]["BACKEND"] = "django.contrib.staticfiles.storage.StaticFilesStorage"  # noqa: F405

ASP_ITOU_PREFIX = "XXXXX"  # same as in our fixtures
ITOU_PROTOCOL = "http"
ITOU_FQDN = "localhost:8000"

DATABASES["default"]["HOST"] = os.getenv("PGHOST", "127.0.0.1")  # noqa: F405
DATABASES["default"]["PORT"] = os.getenv("PGPORT", "5432")  # noqa: F405
DATABASES["default"]["NAME"] = os.getenv("PGDATABASE", "itou")  # noqa: F405
DATABASES["default"]["USER"] = os.getenv("PGUSER", "postgres")  # noqa: F405
DATABASES["default"]["PASSWORD"] = os.getenv("PGPASSWORD", "password")  # noqa: F405

MAILJET_API_KEY_PRINCIPAL = "API_MAILJET_KEY_PRINCIPAL"
MAILJET_SECRET_KEY_PRINCIPAL = "API_MAILJET_SECRET_PRINCIPAL"

AWS_S3_ENDPOINT_URL = f"http://{os.getenv('CELLAR_ADDON_HOST', 'localhost:9000')}/"
AWS_S3_ACCESS_KEY_ID = "minioadmin"
AWS_S3_SECRET_ACCESS_KEY = "minioadmin"
AWS_STORAGE_BUCKET_NAME = "tests"
