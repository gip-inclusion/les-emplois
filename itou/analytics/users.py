from django.db.models import Count

from itou.users import enums as users_enums, models as users_models

from . import models


def collect_analytics_data(before):
    counts = {
        info["kind"]: info["count"]
        for info in users_models.User.objects.values("kind").annotate(count=Count("pk")).values("kind", "count")
    }
    return {
        models.DatumCode.USER_COUNT: sum(counts.values()),
        models.DatumCode.USER_JOB_SEEKER_COUNT: counts.get(users_enums.KIND_JOB_SEEKER, 0),
        models.DatumCode.USER_PRESCRIBER_COUNT: counts.get(users_enums.KIND_PRESCRIBER, 0),
        models.DatumCode.USER_EMPLOYER_COUNT: counts.get(users_enums.KIND_EMPLOYER, 0),
        models.DatumCode.USER_LABOR_INSPECTOR_COUNT: counts.get(users_enums.KIND_LABOR_INSPECTOR, 0),
        models.DatumCode.USER_ITOU_STAFF_COUNT: counts.get(users_enums.KIND_ITOU_STAFF, 0),
    }
