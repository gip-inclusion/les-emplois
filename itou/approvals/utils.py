from itou.companies.enums import SIAE_WITH_CONVENTION_KINDS


def get_user_last_accepted_siae_job_application(user):
    if not user.is_job_seeker:
        return None

    # Some candidates may not have accepted job applications
    # Assuming it's the case can lead to issues downstream
    return (
        user.job_applications.accepted()
        .filter(to_company__kind__in=SIAE_WITH_CONVENTION_KINDS)
        .with_accepted_at()
        .order_by("-accepted_at", "-hiring_start_at")
        .first()
    )


def last_hire_was_made_by_siae(user, siae):
    if not user.is_job_seeker:
        return False
    last_accepted_job_application = get_user_last_accepted_siae_job_application(user)
    return last_accepted_job_application and last_accepted_job_application.to_company_id == siae.pk
