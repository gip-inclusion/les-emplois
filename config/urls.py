from django.conf import settings
from django.contrib import admin
from django.urls import include, path, re_path, register_converter
from django.views.generic import TemplateView

from itou.utils.urls import SiretConverter
from itou.www.dashboard import views as dashboard_views
from itou.www.login import views as login_views
from itou.www.signup import views as signup_views


register_converter(SiretConverter, "siret")

urlpatterns = [
    path("admin/", admin.site.urls),
    # --------------------------------------------------------------------------------------
    # allauth URLs. Order is important because some URLs are overriden.
    # --------------------------------------------------------------------------------------
    # Override allauth `account_signup` URL.
    # /accounts/signup/ <=> account_signup
    # We don't want any user to be able to signup using the default allauth `signup` url
    # because we have multiple specific signup processes for different kind of users.
    re_path(r"^accounts/signup/$", signup_views.signup),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_login` URL.
    # /accounts/login/ <=> account_login
    # We override this view because the login page should look slightly differently
    # for job seekers, prescribers and employers.
    # Also, PEAMU is only available for job seekers.
    re_path(r"^accounts/login/$", login_views.login),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_change_password` URL.
    # /accounts/password/change/ <=> account_change_password
    # https://github.com/pennersr/django-allauth/issues/468
    re_path(r"^accounts/password/change/$", dashboard_views.password_change),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_reset_password` URL.
    # Avoid user enumeration via password reset page.
    re_path(r"^accounts/password/reset/$", signup_views.ItouPasswordResetView.as_view()),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_logout` URL.
    # /accounts/logout/ <=> account_logout
    # We need custom code to process PEAMU logout.
    re_path(r"^accounts/logout/$", dashboard_views.logout),
    # --------------------------------------------------------------------------------------
    # Other allauth URLs.
    path("accounts/", include("allauth.urls")),
    # --------------------------------------------------------------------------------------
    # PEAMU URLs.
    path("accounts/", include("itou.allauth_adapters.peamu.urls")),
    # FranceConnect URLs.
    path("franceconnect/", include("itou.france_connect.urls")),
    # --------------------------------------------------------------------------------------
    # API.
    path("api/v1/", include("itou.api.urls", namespace="v1")),
    path("api/", include("itou.www.api.urls")),
    # www.
    path("", include("itou.www.home.urls")),
    path("apply/", include("itou.www.apply.urls")),
    path("approvals/", include("itou.www.approvals_views.urls")),
    path("autocomplete/", include("itou.www.autocomplete.urls")),
    path("dashboard/", include("itou.www.dashboard.urls")),
    path("employee_record/", include("itou.www.employee_record_views.urls")),
    path("institutions/", include("itou.www.institutions_views.urls")),
    path("invitations/", include("itou.www.invitations_views.urls")),
    path("prescribers/", include("itou.www.prescribers_views.urls")),
    path("search/", include("itou.www.search.urls")),
    path("siae/", include("itou.www.siaes_views.urls")),
    path("signup/", include("itou.www.signup.urls")),
    path("stats/", include("itou.www.stats.urls")),
    path("versions/", include("itou.www.releases.urls")),
    # Enable Anymailâ€™s status tracking
    # https://anymail.readthedocs.io/en/stable/esps/mailjet/#status-tracking-webhooks
    re_path(r"^webhooks/anymail/", include("anymail.urls")),
    path("welcoming_tour/", include("itou.www.welcoming_tour.urls")),
    # Static pages.
    path("accessibility/", TemplateView.as_view(template_name="static/accessibility.html"), name="accessibility"),
    # Errors pages.
    path("403/", TemplateView.as_view(template_name="403.html"), name="403"),
    path("404/", TemplateView.as_view(template_name="404.html"), name="404"),
    path("500/", TemplateView.as_view(template_name="500.html"), name="500"),
]

if settings.DEBUG and "debug_toolbar" in settings.INSTALLED_APPS:
    import debug_toolbar

    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
    # In the local developpemnt environment, we enable the default callback URLs
    urlpatterns = [path("", include("itou.france_connect.urls"))] + urlpatterns
