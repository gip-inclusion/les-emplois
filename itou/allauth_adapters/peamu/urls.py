from django.urls import include, re_path

from itou.allauth_adapters.peamu import views as peamu_views
from itou.allauth_adapters.peamu.provider import PEAMUProvider


def default_urlpatterns(provider):
    urlpatterns = [
        re_path(r"^login/$", peamu_views.oauth2_login, name=provider.id + "_login"),
        re_path(r"^login/callback/$", peamu_views.oauth2_callback, name=provider.id + "_callback"),
    ]

    return [
        re_path(r"^profile/$", peamu_views.redirect_to_dashboard_view),
        re_path(r"^" + provider.get_slug() + r"/", include(urlpatterns)),
    ]


urlpatterns = default_urlpatterns(PEAMUProvider)
