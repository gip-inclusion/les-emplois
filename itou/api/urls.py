from django.urls import include, path
from drf_spectacular.views import SpectacularAPIView, SpectacularRedocView
from rest_framework import routers
from rest_framework.authtoken import views as auth_views

from itou.api.data_inclusion_api.views import DataInclusionStructureView

from .applicants_api.views import ApplicantsView
from .employee_record_api.viewsets import (
    DummyEmployeeRecordViewSet,
    EmployeeRecordUpdateNotificationViewSet,
    EmployeeRecordViewSet,
)
from .siae_api.viewsets import SiaeViewSet


# High level app for API
# https://docs.djangoproject.com/en/dev/topics/http/urls/#url-namespaces-and-included-urlconfs
app_name = "itou.api"

# Using DRF router with viewsets means automatic definition of utl patterns
router = routers.DefaultRouter()
router.register(r"employee-records", EmployeeRecordViewSet, basename="employee-records")
router.register(
    r"employee-record-notifications",
    EmployeeRecordUpdateNotificationViewSet,
    basename="employee-record-notifications",
)
router.register(r"dummy-employee-records", DummyEmployeeRecordViewSet, basename="dummy-employee-records")
router.register(r"siaes", SiaeViewSet, basename="siaes")

urlpatterns = [
    # TokenAuthentication endpoint to get token from login/password.
    path("token-auth/", auth_views.obtain_auth_token, name="token-auth"),
    # Needed for Browseable API (dev)
    path("api-auth/", include("rest_framework.urls", namespace="rest_framework")),
    # OpenAPI
    # See: https://www.django-rest-framework.org/topics/documenting-your-api/
    # OAS 3 YAML schema (downloadable)
    path(
        "oas3/",
        SpectacularAPIView.as_view(),
        name="openapi_schema",
    ),
    path(
        "redoc/",
        SpectacularRedocView.as_view(url_name="v1:openapi_schema"),
        name="redoc",
    ),
]

urlpatterns += router.urls

urlpatterns += [
    path("structures/", DataInclusionStructureView.as_view(), name="structures-list"),
    path("candidats/", ApplicantsView.as_view(), name="applicants-list"),
]
