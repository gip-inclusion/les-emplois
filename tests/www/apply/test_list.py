from dateutil.relativedelta import relativedelta
from django.urls import reverse
from django.utils import timezone
from freezegun import freeze_time

from tests.companies.factories import CompanyFactory
from tests.job_applications.factories import JobApplicationFactory
from tests.prescribers.factories import PrescriberOrganizationWithMembershipFactory
from tests.users.factories import JobSeekerFactory
from tests.utils.test import parse_response_to_soup


@freeze_time("2023-04-13")
def test_list_warns_about_long_awaiting_applications(client, snapshot):
    hit_pit = CompanyFactory(pk=42, name="Hit Pit", with_membership=True)

    now = timezone.now()
    org = PrescriberOrganizationWithMembershipFactory(
        membership__user__first_name="Max", membership__user__last_name="Throughput"
    )
    sender = org.active_members.get()
    job_seeker = JobSeekerFactory(first_name="Jacques", last_name="Henry")
    JobApplicationFactory(
        id="11111111-1111-1111-1111-111111111111",
        to_company=hit_pit,
        job_seeker=job_seeker,
        sender=sender,
        message="Third application",
        created_at=now - relativedelta(weeks=2),
    )
    JobApplicationFactory(
        id="22222222-2222-2222-2222-222222222222",
        to_company=hit_pit,
        job_seeker=job_seeker,
        sender=sender,
        message="Second application",
        created_at=now - relativedelta(weeks=3, days=5),
    )
    JobApplicationFactory(
        id="33333333-3333-3333-3333-333333333333",
        to_company=hit_pit,
        job_seeker=job_seeker,
        sender=sender,
        message="First application",
        created_at=now - relativedelta(weeks=8),
    )

    client.force_login(hit_pit.members.get())
    response = client.get(reverse("apply:list_for_siae"))
    results_section = parse_response_to_soup(response, selector="#job-applications-section")
    assert str(results_section) == snapshot(name="SIAE - warnings for 2222 and 3333")

    client.force_login(sender)
    response = client.get(reverse("apply:list_prescriptions"))
    results_section = parse_response_to_soup(response, selector="#job-applications-section")
    assert str(results_section) == snapshot(name="PRESCRIBER - warnings for 2222 and 3333")

    client.force_login(job_seeker)
    response = client.get(reverse("apply:list_for_job_seeker"))
    results_section = parse_response_to_soup(response, selector="#job-applications-section")
    assert str(results_section) == snapshot(name="JOB SEEKER - no warnings")
