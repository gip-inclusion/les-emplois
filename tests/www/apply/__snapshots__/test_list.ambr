# serializer version: 1
# name: test_list_warns_about_long_awaiting_applications[JOB SEEKER - no warnings]
  '''
  <section aria-labelledby="results" id="job-applications-section">
      <div class="d-flex flex-column flex-md-row align-items-md-center justify-content-md-between mb-3 mb-md-4">
          <h3 class="h4 mb-0" id="results">
              
                  3 <strong>résultats</strong>
              
          </h3>
          <div class="flex-column flex-md-row btn-group btn-group-sm btn-group-action">
              
          </div>
      </div>
      
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 30 mars 2023 par
          
              
                  <strong>Max THROUGHPUT</strong>
              
              
                  <i aria-hidden="true" class="ri-home-smile-2-line me-1"></i>Orienteur
              
          
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_11111111-1111-1111-1111-111111111111">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <span>Entreprise d'insertion</span>
              <a class="h3 d-block text-decoration-none" href="/company/42/card">Hit pit</a>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex justify-content-end">
                          
                          
                              <a class="btn btn-outline-primary" href="/apply/11111111-1111-1111-1111-111111111111/jobseeker/details?back_url=/apply/job_seeker/list">
                                  Voir ma candidature
                              </a>
                          
                      </div>
                  </div>
              </div>
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 18 mars 2023 par
          
              
                  <strong>Max THROUGHPUT</strong>
              
              
                  <i aria-hidden="true" class="ri-home-smile-2-line me-1"></i>Orienteur
              
          
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_22222222-2222-2222-2222-222222222222">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <span>Entreprise d'insertion</span>
              <a class="h3 d-block text-decoration-none" href="/company/42/card">Hit pit</a>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex flex-column flex-md-row justify-content-md-between align-items-md-center">
                          
                              <p class="text-warning fs-sm mb-3 mb-md-0">
                                  <i aria-hidden="true" class="ri-time-line ri-lg me-1"></i>
                                  En attente de réponse depuis 3 semaines.
                              </p>
                          
                          
                              <a class="btn btn-outline-primary btn-block w-100 w-md-auto" href="/apply/22222222-2222-2222-2222-222222222222/jobseeker/details?back_url=/apply/job_seeker/list">
                                  Voir ma candidature
                              </a>
                          
                      </div>
                  </div>
              </div>
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 16 février 2023 par
          
              
                  <strong>Max THROUGHPUT</strong>
              
              
                  <i aria-hidden="true" class="ri-home-smile-2-line me-1"></i>Orienteur
              
          
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_33333333-3333-3333-3333-333333333333">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <span>Entreprise d'insertion</span>
              <a class="h3 d-block text-decoration-none" href="/company/42/card">Hit pit</a>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex flex-column flex-md-row justify-content-md-between align-items-md-center">
                          
                              <p class="text-warning fs-sm mb-3 mb-md-0">
                                  <i aria-hidden="true" class="ri-time-line ri-lg me-1"></i>
                                  En attente de réponse depuis 8 semaines.
                              </p>
                          
                          
                              <a class="btn btn-outline-primary btn-block w-100 w-md-auto" href="/apply/33333333-3333-3333-3333-333333333333/jobseeker/details?back_url=/apply/job_seeker/list">
                                  Voir ma candidature
                              </a>
                          
                      </div>
                  </div>
              </div>
          
          
  
  
  
      
  </section>
  '''
# ---
# name: test_list_warns_about_long_awaiting_applications[PRESCRIBER - warnings for 2222 and 3333]
  '''
  <section aria-labelledby="results" id="job-applications-section">
      <div class="d-flex flex-column flex-md-row align-items-md-center justify-content-md-between mb-3 mb-md-4">
          <h3 class="h4 mb-0" id="results">
              
                  3 <strong>résultats</strong>
              
          </h3>
          <div class="flex-column flex-md-row btn-group btn-group-sm btn-group-action">
              
          </div>
      </div>
      
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 30 mars 2023 par
          
              <strong>Vous</strong>
          
          
              chez
              <a class="btn-ico btn-link" href="/company/42/card?back_url=/apply/prescriptions/list">
                  <i aria-hidden="true" class="ri-community-line font-weight-medium ri-sm me-1"></i>
                  <span>Hit pit</span>
              </a>
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_11111111-1111-1111-1111-111111111111">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <h3>J… H…</h3>
              <div class="d-flex flex-column flex-md-row align-items-start align-items-md-end gap-1 gap-md-2">
                  
                  
                      
                          
                              
                                  <span class="badge badge-xs rounded-pill bg-success-lighter text-success">
                                      <i aria-hidden="true" class="ri-check-line"></i>
                                      Éligible à l’IAE
                                  </span>
                              
                          
                      
                  
              </div>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex justify-content-end">
                          
                          
                              <a aria-label="Gérer la candidature de J… H…" class="btn btn-outline-primary" data-matomo-action="clic" data-matomo-category="candidature" data-matomo-event="true" data-matomo-option="voir_candidature_prescripteur" href="/apply/11111111-1111-1111-1111-111111111111/prescriber/details?back_url=/apply/prescriptions/list">
                                  Voir sa candidature
                              </a>
                          
                      </div>
                  </div>
              </div>
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 18 mars 2023 par
          
              <strong>Vous</strong>
          
          
              chez
              <a class="btn-ico btn-link" href="/company/42/card?back_url=/apply/prescriptions/list">
                  <i aria-hidden="true" class="ri-community-line font-weight-medium ri-sm me-1"></i>
                  <span>Hit pit</span>
              </a>
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_22222222-2222-2222-2222-222222222222">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <h3>J… H…</h3>
              <div class="d-flex flex-column flex-md-row align-items-start align-items-md-end gap-1 gap-md-2">
                  
                  
                      
                          
                              
                                  <span class="badge badge-xs rounded-pill bg-success-lighter text-success">
                                      <i aria-hidden="true" class="ri-check-line"></i>
                                      Éligible à l’IAE
                                  </span>
                              
                          
                      
                  
              </div>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex flex-column flex-md-row justify-content-md-between align-items-md-center">
                          
                              <p class="text-warning fs-sm mb-3 mb-md-0">
                                  <i aria-hidden="true" class="ri-time-line ri-lg me-1"></i>
                                  En attente de réponse depuis 3 semaines.
                              </p>
                          
                          
                              <a aria-label="Gérer la candidature de J… H…" class="btn btn-outline-primary btn-block w-100 w-md-auto" data-matomo-action="clic" data-matomo-category="candidature" data-matomo-event="true" data-matomo-option="voir_candidature_prescripteur" href="/apply/22222222-2222-2222-2222-222222222222/prescriber/details?back_url=/apply/prescriptions/list">
                                  Voir sa candidature
                              </a>
                          
                      </div>
                  </div>
              </div>
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 16 février 2023 par
          
              <strong>Vous</strong>
          
          
              chez
              <a class="btn-ico btn-link" href="/company/42/card?back_url=/apply/prescriptions/list">
                  <i aria-hidden="true" class="ri-community-line font-weight-medium ri-sm me-1"></i>
                  <span>Hit pit</span>
              </a>
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_33333333-3333-3333-3333-333333333333">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <h3>J… H…</h3>
              <div class="d-flex flex-column flex-md-row align-items-start align-items-md-end gap-1 gap-md-2">
                  
                  
                      
                          
                              
                                  <span class="badge badge-xs rounded-pill bg-success-lighter text-success">
                                      <i aria-hidden="true" class="ri-check-line"></i>
                                      Éligible à l’IAE
                                  </span>
                              
                          
                      
                  
              </div>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex flex-column flex-md-row justify-content-md-between align-items-md-center">
                          
                              <p class="text-warning fs-sm mb-3 mb-md-0">
                                  <i aria-hidden="true" class="ri-time-line ri-lg me-1"></i>
                                  En attente de réponse depuis 8 semaines.
                              </p>
                          
                          
                              <a aria-label="Gérer la candidature de J… H…" class="btn btn-outline-primary btn-block w-100 w-md-auto" data-matomo-action="clic" data-matomo-category="candidature" data-matomo-event="true" data-matomo-option="voir_candidature_prescripteur" href="/apply/33333333-3333-3333-3333-333333333333/prescriber/details?back_url=/apply/prescriptions/list">
                                  Voir sa candidature
                              </a>
                          
                      </div>
                  </div>
              </div>
          
          
  
  
  
      
  </section>
  '''
# ---
# name: test_list_warns_about_long_awaiting_applications[SIAE - warnings for 2222 and 3333]
  '''
  <section aria-labelledby="results" id="job-applications-section">
      <div class="d-flex flex-column flex-md-row align-items-md-center justify-content-md-between mb-3 mb-md-4">
          <h3 class="h4 mb-0" id="results">
              
                  3 <strong>résultats</strong>
              
          </h3>
          <div class="flex-column flex-md-row btn-group btn-group-sm btn-group-action">
              
          </div>
      </div>
      
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 30 mars 2023 par
          
              
                  <strong>Max THROUGHPUT</strong>
              
              
                  <i aria-hidden="true" class="ri-home-smile-2-line me-1"></i>Orienteur
              
          
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_11111111-1111-1111-1111-111111111111">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <h3>Jacques HENRY</h3>
              <div class="d-flex flex-column flex-md-row align-items-start align-items-md-end gap-1 gap-md-2">
                  
                  
                      
                          
                              
                                  <span class="badge badge-xs rounded-pill bg-success-lighter text-success">
                                      <i aria-hidden="true" class="ri-check-line"></i>
                                      Éligible à l’IAE
                                  </span>
                              
                          
                      
                  
              </div>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex justify-content-end">
                          
                          
                              <a aria-label="Gérer la candidature de Jacques HENRY" class="btn btn-outline-primary" data-matomo-action="clic" data-matomo-category="candidature" data-matomo-event="true" data-matomo-option="voir-candidature-employeur" href="/apply/11111111-1111-1111-1111-111111111111/siae/details?back_url=/apply/siae/list">Voir sa candidature</a>
                          
                      </div>
                  </div>
              </div>
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 18 mars 2023 par
          
              
                  <strong>Max THROUGHPUT</strong>
              
              
                  <i aria-hidden="true" class="ri-home-smile-2-line me-1"></i>Orienteur
              
          
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_22222222-2222-2222-2222-222222222222">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <h3>Jacques HENRY</h3>
              <div class="d-flex flex-column flex-md-row align-items-start align-items-md-end gap-1 gap-md-2">
                  
                  
                      
                          
                              
                                  <span class="badge badge-xs rounded-pill bg-success-lighter text-success">
                                      <i aria-hidden="true" class="ri-check-line"></i>
                                      Éligible à l’IAE
                                  </span>
                              
                          
                      
                  
              </div>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex flex-column flex-md-row justify-content-md-between align-items-md-center">
                          
                              <p class="text-warning fs-sm mb-3 mb-md-0">
                                  <i aria-hidden="true" class="ri-time-line ri-lg me-1"></i>
                                  En attente de réponse depuis 3 semaines.
                              </p>
                          
                          
                              <a aria-label="Gérer la candidature de Jacques HENRY" class="btn btn-outline-primary btn-block w-100 w-md-auto" data-matomo-action="clic" data-matomo-category="candidature" data-matomo-event="true" data-matomo-option="voir-candidature-employeur" href="/apply/22222222-2222-2222-2222-222222222222/siae/details?back_url=/apply/siae/list">Voir sa candidature</a>
                          
                      </div>
                  </div>
              </div>
          
              <div class="c-box c-box--results has-links-inside my-3 my-md-4">
                  
                      
  
  <div class="c-box--results__header">
      
  
  <div class="d-flex flex-column flex-lg-row gap-1 gap-lg-3 mb-3">
      <p class="fs-sm mb-0 flex-grow-1">
          Émise le 16 février 2023 par
          
              
                  <strong>Max THROUGHPUT</strong>
              
              
                  <i aria-hidden="true" class="ri-home-smile-2-line me-1"></i>Orienteur
              
          
          
      </p>
      <div><span class="badge rounded-pill text-nowrap badge-sm mb-1 bg-info" id="state_33333333-3333-3333-3333-333333333333">Nouvelle candidature</span></div>
  </div>
  
      <div class="c-box--results__summary">
          <i aria-hidden="true" class="ri-user-line"></i>
          <div>
              <h3>Jacques HENRY</h3>
              <div class="d-flex flex-column flex-md-row align-items-start align-items-md-end gap-1 gap-md-2">
                  
                  
                      
                          
                              
                                  <span class="badge badge-xs rounded-pill bg-success-lighter text-success">
                                      <i aria-hidden="true" class="ri-check-line"></i>
                                      Éligible à l’IAE
                                  </span>
                              
                          
                      
                  
              </div>
          </div>
      </div>
  </div>
  <hr class="m-0"/>
  <div class="c-box--results__body">
      <div class="c-box--results__list-data">
          
          
  
  
      
          <div class="c-info c-info--secondary">
              <span class="c-info__summary">Candidature spontanée</span>
          </div>
      
  
  
      </div>
  </div>
  
                  
                  <div class="c-box--results__footer">
                      <div class="d-flex flex-column flex-md-row justify-content-md-between align-items-md-center">
                          
                              <p class="text-warning fs-sm mb-3 mb-md-0">
                                  <i aria-hidden="true" class="ri-time-line ri-lg me-1"></i>
                                  En attente de réponse depuis 8 semaines.
                              </p>
                          
                          
                              <a aria-label="Gérer la candidature de Jacques HENRY" class="btn btn-outline-primary btn-block w-100 w-md-auto" data-matomo-action="clic" data-matomo-category="candidature" data-matomo-event="true" data-matomo-option="voir-candidature-employeur" href="/apply/33333333-3333-3333-3333-333333333333/siae/details?back_url=/apply/siae/list">Voir sa candidature</a>
                          
                      </div>
                  </div>
              </div>
          
          
  
  
  
      
  </section>
  '''
# ---
